### DISCUSSION TITLE
IKE session keys

### DISCUSSION THREAD

**Discussion by hnsantu554** [strongswan/strongswan#2469] [General]
I am using strongswan on linux system , how ike session keys and AH/ESP session keys are managed?
Is it possible to store these keys in any secured storage and access it from there ?
Please help me to understand thank you

**Comment by tobiasbrunner** [comment]
What kind of secure storage are you thinking about? Also, IPsec is typically handled by the Linux kernel, so once they are configured, strongSwan doesn't store them anymore. Anyway, you might want to look into the [Trusted Key Manager (TKM)](https://www.codelabs.ch/tkm/) concept.

**Reply by hnsantu554** [reply]
Sure i will check , thanks for the information.
Does strongswan configure AH/ESP session keys into Linux kernel during IKE operations?
 Please help me to understand, thanks in advance

**Reply by tobiasbrunner** [reply]
> Does strongswan configure AH/ESP session keys into Linux kernel during IKE operations?

Yes.
